runner/README.md

```diff
@@ -1,11 +1,57 @@
-Runner is a library for running Scoot processes.
+# Runner Design #
+package runner is structs and interfaces to run Commands in Scoot. package runners contains implementations. This doc describes both.
 
-The top-level interface is Runner. It's a system that runs processes against snapshots and lets you query their status.
+## Summary ##
+* Service (interface) combines Controller and StatusReader
+* Controller (interface) controls when to Run a Command.
+  * Controller calls Invoker
+  * Controller writes to StatusWriter
+* runners.Invoker (struct) runs one command using Execer and snapshot.Filer
+  * implements Abort and Timeout
+* runners.Statuses (struct) offers read and write of RunStatus
+  * satisfies interfaces StatusWriter and StatusReader
 
-There's an implementation in local simpleRunner. This will run at most process at a time.
+## Invoker (runner/runners/invoker.go) ##
+Invoker runs a Scoot command. This involves:
+* setup (check out a snapshot)
+* post-processing (save output)
+* babysitting (implement timeout and abort)
+* streaming RunStatus'es over the course of a run
 
-Underneath Runner is Execer. It is just to run a Unix process locally in a directory. It should be free of Scoot abstractions.
+Conceptually, it lowers a Scoot Command from a high-level abstraction (Snapshots, etc.) to an Execer command (about directories that are paths in the local filesystem). Concretely,  it checks out the input snapshot's files, then uses an Execer to run the command. When the command has finished, it created a snapshot of the command's output.
 
-There is a SimExecer that simulates behavior based on the args passed to it. This lets a caller of a Runner script the behavior of the Execer.
+## Controller (runner/controller.go) ##
+Controller controls the Runs, starting a new one (Run()) or ending one (Abort()). It implements:
+* tenancy (run one at a time? Or multiple? Fixed, or based on system utilization?)
+* queueing (should we accept work we can't start right now)
 
-There will be an Execer that calls os/exec to exec.
\ No newline at end of file
+When a Controller wants to start a Run of a Command, it will use an Invoker.
+
+For now we have one implementation, QueueController
+
+## Statuses interfaces (runner/statuses.go) ##
+There are several interfaces having to do with reading/writing RunStatus'es.
+
+StatusWriter for writing Statuses.
+
+There are more reader interfaces:
+* StatusReader includes all the ones below
+* LegacyStatusReader offers legacy APIs
```

Comment:
Somewhat off topic, but I think we should be very aggressive in deleting all vestigial traces of legacy APIs quickly. Cleaning this out will never be easier than it is now.  I don't remember exactly what this is supporting - closed source binaries?